### PR TITLE
Fixing some minor issues

### PR DESCRIPTION
**Pull request by shahnami**
This PR fixes some minor issues identified during testing, as well as the following:

- https://github.com/OpenZeppelin/defender-serverless/issues/17
- https://github.com/OpenZeppelin/defender-serverless/issues/20
- https://github.com/OpenZeppelin/defender-serverless/issues/16
- https://github.com/OpenZeppelin/defender-serverless/issues/15

- Fixes typing for Autotask Triggers
- Introduces an additional check to validate compatible block watchers when constructing a sentinel
- Introduces global and stack-based secrets
- Clarification in README around resource deletions, types and other caveats.

### CODE REVIEW

File: src/cmd/deploy.ts [hunk 1]
```diff
@@ -217,19 +238,25 @@ export default class DefenderDeploy {
...
-      secrets,
+      allSecrets,
       retrieveExisting,
       // on update
-      async (secret: YSecret, match: any) => {
+      async (secret: YSecret, match: string) => {
+        const stackName = getStackName(this.serverless);
+        const [_, value] = Object.entries(secret)[0];
+        const stackSecretMatch = stackSecrets[match];
+        // if secret key matches that of any in stackSecrets, prepend the key with stackName
```
Comment (spalladino) [inline src/cmd/deploy.ts:257]: What happens if I define two secrets with the same name (eg "foo"), one global and one stack? Wouldn't this consider them both as stack?

File: src/cmd/deploy.ts [hunk 2]
```diff
@@ -243,9 +270,13 @@ export default class DefenderDeploy {
         };
       },
       // on create
-      async (secret: YSecret, stackResourceId: string) => {
+      async (secret: YSecret, _: string) => {
+        const stackName = getStackName(this.serverless);
+        const [key, value] = Object.entries(secret)[0];
+        const stackSecretMatch = stackSecrets[key];
+        // if secret key matches that of any in stackSecrets, prepend the key with stackName
```
Comment (spalladino) [inline src/cmd/deploy.ts:277]: Same as above

**Review by spalladino — APPROVED** [review]

File: src/cmd/deploy.ts (hunk 1 above)
Comment (shahnami) [reply src/cmd/deploy.ts]: Ha, yes :) Nice catch.

File: README.md [hunk 3]
```diff
@@ -84,6 +84,29 @@ The `stackName` (e.g. mystack) is combined with the resource key (e.g. relayer-1
...
+  # optional - global secrets are not affected by stackName changes
+  global:
+    foo: ${self:custom.config.secrets.foo}
+    hello: ${self:custom.config.secrets.hello}
+  # optional - stack secrets (formatted as <stackname>_<secretkey>)
+  stack:
+    test: ${self:custom.config.secrets.test}
+```
+
+### Types and Schema validation
```
[the remark this answers is not in the document]
Comment (shahnami) [reply README.md:104]: Will turn this into an issue, and look into it :)

File: src/cmd/deploy.ts (hunk 1 above)
Comment (shahnami) [reply src/cmd/deploy.ts]: Fixed in [this](https://github.com/OpenZeppelin/defender-serverless/pull/23/commits/f440449d8aec2296f6014fcc14ef10c4dc797715) commit.

**Pull request merged**